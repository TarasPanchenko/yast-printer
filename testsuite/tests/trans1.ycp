/**
 * File:
 *  trans1.ycp
 *
 * Module:
 *  Printer configurator
 *
 * Summary:
 *  Tests for basic translation functions.
 *
 * Authors:
 *  Petr Blahos <blahos@suse.cz>, 2001
 *
 * $Id$
 *
 * Tests simple translation functions.
 */
{
    import "Pkg";

    global define UI::OpenDialog(term t)``{}
    global define UI::CloseDialog()``{}
    global define term UI::UserInput()``{return `lprng ;}

    include "testsuite.ycp";
    map I_READ = $[
	"target" : $[
	    "size" : -1,
	    "tmpdir" : "/tmp/",
	    "yast2" : nil,
	],
	"sysconfig" : $[
	    "console" : $[
		"CONSOLE_FONT" : "",
		"CONSOLE_SCREENMAP" : "",
		"CONSOLE_UNICODEMAP" : "",
		"CONSOLE_MAGIC" : "",
		"CONSOLE_ENCODING" : "",
	    ],
	    "language" : $[
		"RC_LANG" : "",
		"DEFAULT_LANGUAGE" : "",
	    ],
	],
	"etc" : $[
	    "install_inf" : $[
		"Cmdline" : "",
		"InstMode" : nil,
	    ],
	],
	"probe" : $[
	    "architecture" : "i386",
	    "has_apm" : true,
	    "has_pcmcia" : false,
	    "has_smp" : false,
	    "system" : [],
	    "memory" : [],
	    "cpu" : [],
	    "cdrom" : $[
		"manual" : [],
	    ],
	    "floppy" : $[
		"manual" : [],
	    ],
	],
    ];
    map I_WRITE = $[];
    map I_EXEC = $[
        "target" : $[
            "bash_output" : $[],
        ],
    ];
    TESTSUITE_INIT ([I_READ, I_WRITE, I_EXEC], 0);
    include "printer/misc.ycp";
    TESTSUITE_CLEAN ();

    TEST (``(filter1284 ("abcdefghijklmnopqrstuvwxyz0123456789_-+")), [], false);
    TEST (``(filter1284 ("!@#$%^&*()")), [], false);
    TEST (``(filter1284 ("text with unallowed characters: \"/[]{}.'`")), [], false);
}
