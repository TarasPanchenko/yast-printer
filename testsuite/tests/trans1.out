Read	.target.size "/etc/install.inf" -1
Read	.sysconfig.console.CONSOLE_ENCODING ""
Execute	.target.bash_output "locale -k charmap" $[]
Read	.probe.architecture "i386"
Read	.probe.has_apm true
Read	.probe.has_pcmcia false
Read	.probe.has_smp false
Read	.probe.system []
Read	.target.tmpdir "/tmp/"
Return	abcdefghijklmnopqrstuvwxyz0123456789_+
Return	
Return	textwithunallowedcharacters/
